minigalaxy/filesys_utils.py

```diff
@@ -0,0 +1,174 @@
+import subprocess
+import os
+import shutil
+import json
+from minigalaxy.paths import CONFIG_DIR, CACHE_DIR, CONFIG_FILE_PATH, DEFAULT_INSTALL_DIR
+
+
+def _get_white_list():
+    json_config = {"install_dir": DEFAULT_INSTALL_DIR}
+    if os.path.exists(CONFIG_FILE_PATH):
+        with open(CONFIG_FILE_PATH, "r") as file:
+            try:
+                json_config = json.loads(file.read())
+            except json.decoder.JSONDecodeError:
+                pass
+    return [json_config["install_dir"], CONFIG_DIR, CACHE_DIR]
+
+
+def _get_black_list():
+    important_user_dirs = ["", "DESKTOP", "DOCUMENTS", "DOWNLOAD", "MUSIC", "PICTURES", "PUBLICSHARE", "TEMPLATES",
+                           "VIDEOS"]
+    black_list = []
+    for important_dir in important_user_dirs:
+        black_list.append(subprocess.check_output(['xdg-user-dir', important_dir]).decode("utf-8").strip())
+    return black_list
+
+
+def _copy_move_and_overwrite(source, target, copy_or_move=""):
+    err_msg = ""
+    source = os.path.abspath(source)
+    target = os.path.abspath(target)
+    for src_dir, dirs, files in os.walk(source):
+        destination_dir = src_dir.replace(source, target, 1)
+        if not os.path.exists(destination_dir):
+            os.makedirs(destination_dir)
+        for src_file in files:
+            file_to_copy = os.path.join(src_dir, src_file)
+            dst_file = os.path.join(destination_dir, src_file)
+            if copy_or_move == "copy":
+                shutil.copy2(file_to_copy, dst_file)
+            elif copy_or_move == "move":
+                shutil.move(file_to_copy, dst_file)
+            else:
+                err_msg = "Unknown operation: {}".format(copy_or_move)
+                break
+    return err_msg
+
+
+def _check_if_accordance_with_lists(target):
+    err_msd = ""
+    target = os.path.abspath(target)
+    if not target:
+        err_msd = "Operation on no file was requested."
+    inside_white_list = False
+    for approve_dir in _get_white_list():
+        if target.startswith(approve_dir):
+            inside_white_list = True
+            break
+    is_blacklisted = False
+    for deny_dir in _get_black_list():
+        if target.strip() == deny_dir:
+            is_blacklisted = True
+            break
+    if not inside_white_list:
+        err_msd = "{} is not inside white list for file operations.".format(target)
+    elif is_blacklisted:
+        err_msd = "{} is on black list for file operations.".format(target)
+    return err_msd
+
+
+def _check_if_ok_for_copy_or_move(source="", target="", recursive=False, overwrite=False):
```

Comment:
For private functions I would use double underscore.

Reply:
Such a change does produce issues for unittests. Do we really need this?